### PR TITLE
Feature/#159 follow player

### PR DESCRIPTION
**Pull request by Geosearchef**
still needs a bit of tweaking, see review

### CODE REVIEW

File: src/main/java/com/faforever/client/player/FollowedPlayerJoinedGameNotifier.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        i18n.get("followed.joinedGameNotification.title", player.getUsername(), game.getTitle()),
+        i18n.get("followed.joinedGameNotification.action"),
+        IdenticonUtil.createIdenticon(player.getId()),
+        event1 -> future.cancel(true))
+    );
+
+    new Thread(() -> noCatch(() -> {
+      Thread.sleep(3000);
+      future.complete(game.getStatus() == GameStatus.OPEN);
+    })).start();
```
Comment (Geosearchef) [inline src/main/java/com/faforever/client/player/FollowedPlayerJoinedGameNotifier.java:69]: any way to implement this more clean? Use CompleteableFuture.supplyAsync? Or should I just add a new Task and use TaskService?

File: src/test/java/com/faforever/client/player/FollowedPlayerJoinedGameNotifierTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    ArgumentCaptor<TransientNotification> captor = ArgumentCaptor.forClass(TransientNotification.class);
+    verify(notificationService).addNotification(captor.capture());
+
+    TransientNotification notification = captor.getValue();
+    assertThat(notification.getTitle(), is("following junit"));
+    assertThat(notification.getText(), is("Click to cancel"));
+    assertThat(notification.getImage(), notNullValue());
+
+    NoCatch.noCatch(() -> Thread.sleep(3500));//TODO shorten
```
Comment (Geosearchef) [inline src/test/java/com/faforever/client/player/FollowedPlayerJoinedGameNotifierTest.java:76]: really bad, using mocked taskService I could just verify for call/capture the task given as parameter and execute it?

File: src/test/java/com/faforever/client/player/FollowedPlayerJoinedGameNotifierTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    instance.onFollowedJoinedGame(new FollowedPlayerJoinedGameEvent(player, game));
+
+    ArgumentCaptor<TransientNotification> captor = ArgumentCaptor.forClass(TransientNotification.class);
+    verify(notificationService).addNotification(captor.capture());
+
+    TransientNotification notification = captor.getValue();
+    notification.getActionCallback().call(null);
+
+    NoCatch.noCatch(() -> Thread.sleep(3500));//TODO shorten
```
Comment (Geosearchef) [inline src/test/java/com/faforever/client/player/FollowedPlayerJoinedGameNotifierTest.java:97]: really bad, using mocked taskService I could just verify for call/capture the task given as parameter and execute it?

File: src/main/resources/i18n/messages_de_DE.properties [hunk 4]
```diff
@@ -23,25 +24,25 @@ ranked1v1.you=Du
 ranked1v1.players=Spieler (mit mindestens {0} Spielen)
 ranked1v1.play=Spielen
 ranked1v1.cancel=Abbrechen
-ranked1v1.choseFactions.title=W�hle eine Fraktion
-ranked1v1.choseFactions.description=Sobald ein Spiel startet, wird dir eine der ausgew�hlten Fraktionen zugeteilt.
+ranked1v1.choseFactions.title=W�hle eine Fraktion
```
Comment (Geosearchef) [inline src/main/resources/i18n/messages_de_DE.properties]: have to fix the encoding, don't know how that happened

**Comment by 1-alex98** [comment]
@Geosearchef pls update

**Comment by Geosearchef** [comment]
update what?

**Comment by 1-alex98** [comment]
Like rebase or merge develop into it

**Comment by 1-alex98** [comment]
It has conflicts

**Comment by Geosearchef** [comment]
This is a merge festival.

**Comment by Geosearchef** [comment]
rebased

**Comment by 1-alex98** [comment]
Test still failing

**Pull request closed without merging**